library/test/src/lib.rs

```diff
@@ -470,7 +470,17 @@ pub fn run_test(
         let supports_threads = !cfg!(target_os = "emscripten") && !cfg!(target_arch = "wasm32");
         if concurrency == Concurrent::Yes && supports_threads {
             let cfg = thread::Builder::new().name(name.as_slice().to_owned());
-            cfg.spawn(runtest).unwrap();
+            match cfg.spawn(runtest) {
+                Ok(_) => {}
+                Err(err @ io::Error { .. }) if err.kind() == io::ErrorKind::WouldBlock => {
+                    // If spawning a new thread would block, and blocking was
+                    // requested to not occur, just run the test synchronously.
```

Comment:
I was just copying the `WouldBlock` docs. Do we always request no blocking?

Reply:
'no blocking' is not relevant here (https://github.com/rust-lang/rust/issues/46345). The man page for `fork` says:
```
       EAGAIN A  system-imposed  limit  on  the number of threads was encountered.
       EAGAIN The  caller  is operating under the SCHED_DEADLINE scheduling policy
              and does not have the reset-on-fork flag set.  See sched(7).
```
I think we can safely assume this is not a real-time process, so the comment should say
```suggestion
                    // If we're at the thread limit, just run all tests in a single thread.
```